udata/utils.py

```diff
@@ -202,10 +205,29 @@ def recursive_get(obj, key):
 
 
 def unique_string(length=UUID_LENGTH):
-    '''Generate unique string'''
+    '''Generate a unique string'''
     # We need a string at least as long as length
     string = str(uuid4()) * int(math.ceil(length / float(UUID_LENGTH)))
     return string[:length] if length else string
 
 
+class UDataProvider(BaseProvider):
+    '''
+    A Faker provider for UData missing requirements.
+
+    Might be conributed to upstream Faker project
+    '''
+    def unique_string(self, length=UUID_LENGTH):
```

Comment:
mmh I see, you'll have to deal with `self` with a partial and add extra complexity. Let's keep it as is.